python/dataset.cpp

```diff
@@ -248,11 +248,18 @@ void init_dataset(py::module &m) {
   bind_binary<Dataset>(dataset);
   bind_binary<DatasetProxy>(dataset);
   bind_binary<DataProxy>(dataset);
+  bind_binary<Variable>(dataset);
   bind_binary<VariableConstProxy>(dataset);
   bind_binary<Dataset>(datasetProxy);
   bind_binary<DatasetProxy>(datasetProxy);
   bind_binary<DataProxy>(datasetProxy);
+  bind_binary<Variable>(datasetProxy);
   bind_binary<VariableConstProxy>(datasetProxy);
+  bind_binary<Dataset>(dataArray);
+  bind_binary<DatasetProxy>(dataArray);
+  bind_binary<DataProxy>(dataArray);
+  bind_binary<Variable>(dataArray);
+  bind_binary<VariableConstProxy>(dataArray);
 
```

Comment:
Are all these really necessary? Most cases should just be handled by implicit conversions, e.g., `Variable` -> `VariableConstProxy`.